### ISSUE TITLE
Re-evaluate legacy spoiler tests

### ISSUE THREAD

**Issue by taldcroft** [sot/proseco#220]
From https://github.com/sot/proseco/issues/203#issuecomment-450416918

> The issue with using the full machinery is that check_spoil_contrib and has_spoiler_in_box and check_mag_spoilers all overlap.

That begs the question of whether they are all necessary and helpful, and having this conversation is a good start to getting me to review the code more closely and think about the checks.  :smile:  As a preface, it is understood the plan for the first MATLAB-proseco release was to start by adopting key parts of the SAUSAGE algorithm fairly closely, and we are doing that.  With that...

Looking at `has_spoiler_in_box`, that seems like a relic of when we didn't have a better way.  I'd be perfectly happy taking a 10.0 mag guide star that has a 13.95 mag spoiler which is 1 arcsec away since this will have only a beneficial effect (making the guide star just a bit brighter with almost no centroid disturbance).

Likewise `check_mag_spoilers` has a long heritage but I'm not sure if it is really what we want going forward.  I don't know the basis for the characteristic numbers or the linear distance - dmag relationship.

**Comment by jeanconn** [comment]
Indeed.   The check_mag_spoilers test is the one with the long history from SAUSAGE, but the inner boundary has been removed.  But removing the inner boundary doesn't help much with has_spoiler_in_box and check_spoil_contrib in place.  IIRC has_spoiler_in_box and check_spoil_contrib were both implemented directly as suggestions out of our more recent meetings; how they interact and what's really required/reasonable are different questions.

**Comment by jeanconn** [comment]
Then again, I have no idea what our timetable is for proseco improvements; I thought it would be 6 months before we had a chance to change the algorithm.

**Comment by taldcroft** [comment]
We can change the algorithm in 4.4 any time...  When we can release it is another question.

**Comment by taldcroft** [comment]
See also 7f28dce and related comments starting from https://github.com/sot/proseco/pull/218#issuecomment-450862760.

**Comment by taldcroft** [comment]
Takeaway from #218 is that the current algorithm is *conservative* (which means it is OK for promotion and operational use), and that's where I was going in this issue.
